airbyte-cdk/bulk/toolkits/extract-jdbc/src/main/kotlin/io/airbyte/cdk/read/JdbcPartitionsCreator.kt

```diff
@@ -37,6 +36,14 @@ sealed class JdbcPartitionsCreator<
     fun interface AcquiredResources : AutoCloseable
 
     override fun tryAcquireResources(): PartitionsCreator.TryAcquireResourcesStatus {
+        if (configuration.global) {
+            for (global in stateQuerier.feeds.filterIsInstance<Global>()) {
+                // When CDC is involved, wait for CDC to make progress before snapshotting.
+                if (stateQuerier.current(global) == null) {
+                    return PartitionsCreator.TryAcquireResourcesStatus.RETRY_LATER
+                }
+            }
+        }
```

Comment:
Instead of a resource I plumbed the `StateQuerier` in here and use that to decide if the snapshot needs to wait or not.